### PR TITLE
fix: round custom average metrics

### PR DESCRIPTION
**Pull request by owlas**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #1924 

### Description:

Works in explorer + chart. Still not formatting tooltips in charts (see #1727)

### Preview:

Screenshot 2022-04-28 at 15 40 21

Screenshot 2022-04-28 at 15 41 07

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

**Comment by owlas** [comment]
@ZeRego there's a lot of places in the code where it'd be nice to `get(explore, query)` and it returns all fields with both. I've just implemented this in the fastest way possible.

File: packages/common/src/index.ts [hunk 1]
```diff
@@ -1225,27 +1225,41 @@ export function formatValue<T>(
...
 
-export function getFieldMap(explore: Explore): Record<string, CompiledField> {
+export function getFieldMap(
+    explore: Explore,
+    additionalMetrics: AdditionalMetric[] = [],
+): Record<string, Pick<CompiledField, 'format' | 'round'>> {
+    const additionalMetricFields = additionalMetrics.reduce(
+        (acc, additionalMetric) => ({
+            ...acc,
+            [fieldId(additionalMetric)]: {
```
Comment (rephus) [inline packages/common/src/index.ts:1235]: so you are applying the same format as the original field of the original metric, right ?

File: packages/frontend/src/components/TableTree.tsx [hunk 2]
```diff
@@ -175,6 +175,7 @@ const NodeItemButtons: FC<{
                     dimension.label
                 } on the table ${dimension.tableLabel}`,
                 type,
+                ...(type === MetricType.AVERAGE ? { round: 2 } : {}),
```
Comment (rephus) [inline packages/frontend/src/components/TableTree.tsx:178]: hardcoded round for the table, :+1:

File: packages/frontend/src/hooks/useEcharts.ts [hunk 3]
```diff
@@ -142,7 +142,9 @@ export const getEchartsSeries = (
     originalData: ApiQueryResults['rows'],
     cartesianChart: CartesianChart,
     pivotKey: string | undefined,
-    formats: Record<string, CompiledField> | undefined,
+    formats:
```
Comment (rephus) [inline packages/frontend/src/hooks/useEcharts.ts:145]: :see_no_evil:

**Review by rephus — COMMENTED** [review]
LGTM

**Review by rephus — APPROVED** [review]

**Review by ZeRego — CHANGES_REQUESTED** [review]
Just need a small cleanup.

**Comment by rephus** [comment]
Should this be rounded ? 
image

**Comment by rephus** [comment]
It seems there are no dimensions with round/format, I'll try it locally

**Comment by rephus** [comment]
image


image

I was expecting all custom metrics to have € and round...

**Comment by owlas** [comment]
@rephus that feels like a much better solution than what we have now!!!

This ticket was for "applying `round: 2` to every average metric" - but we could open another ticket for "applying dimension formatting to the custom metric".

File: packages/common/src/index.ts (hunk 1 above)
Comment (owlas) [reply packages/common/src/index.ts:1235]: No! The `additionalMetrics` have unique ids. It's just `{table}_{name}`

File: packages/common/src/index.ts [hunk 4]
```diff
@@ -1225,27 +1225,41 @@ export function formatValue<T>(
...
+    );
     return getFields(explore).reduce(
         (sum, field) => ({
             ...sum,
             [fieldId(field)]: { format: field.format, round: field.round },
         }),
-        {},
+        additionalMetricFields,
     ); // e.g { 'my_table_my_dimension': {format: 'usd', round: 1} }}
 }
```
[the remark this answers is not in the document]
Comment (owlas) [reply packages/common/src/index.ts:1249]: Thank you, what a clown 🤡

File: packages/frontend/src/components/TableTree.tsx (hunk 2 above)
Comment (owlas) [reply packages/frontend/src/components/TableTree.tsx:178]: Yep as per the ticket but your suggestion is actually much better

File: packages/frontend/src/hooks/useEcharts.ts (hunk 3 above)
Comment (owlas) [reply packages/frontend/src/hooks/useEcharts.ts:145]: Haha it's ok we have this all over the codebase. But it makes it much more flexible if we only pick what we need

**Review by ZeRego — APPROVED** [review]

**Pull request merged**